src/Symfony/Component/Cache/Traits/AbstractAdapterTrait.php

```diff
@@ -364,6 +364,10 @@ private function generateItems(iterable $items, array &$keys): \Generator
 
     private function getId($key)
     {
+        if (count($this->ids) > 1000) {
```

Comment:
I'd suggest moving this after this line:
![image](https://user-images.githubusercontent.com/243674/141434566-98fae4e7-7fbd-4cf2-b172-cc0ad7249829.png)

Please also add a leading `\` to `\count()`